internal/api/v1beta1/resource.go

```diff
@@ -82,6 +97,32 @@ func (h Handler) CreateResource(ctx context.Context, request *shieldv1beta1.Crea
 		}
 	}
 
+	relations := request.GetBody().GetRelations()
+	for _, r := range relations {
+		subject := strings.Split(r.Subject, ":")
+		if len(subject) != 2 {
+			logger.Error(fmt.Sprintf("inadequate subject format: %s", r.Subject))
+			continue
+		}
+
+		_, err := h.relationService.Create(ctx, relation.RelationV2{
+			Object: relation.Object{
+				ID:          newResource.Idxa,
+				NamespaceID: newResource.NamespaceID,
+			},
+			Subject: relation.Subject{
+				RoleID:    r.RoleName,
```

Comment:
role is something that is part of object definition. Any reason why we have it bundled with subject?

Reply:
Because `subject` is being assigned a `role`